### PR TITLE
[449] Move navigation bar to Material-UI

### PR DESCRIPTION
**Pull request by gcoutable**
### Type of this PR 

- [X] Bug fix
- [ ] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)

https://github.com/eclipse-sirius/sirius-components/issues/449

### What does this PR do?

It moves the navigation bar to material-ui

### CODE REVIEW

File: frontend/src/icons/SiriusIcon.tsx [hunk 1]
```diff
@@ -10,23 +10,20 @@
...
-    <svg
+    <SvgIcon
       xmlns="http://www.w3.org/2000/svg"
       viewBox="0 0 24 25"
       aria-labelledby="title"
       aria-describedby="desc"
       role="img"
       {...props}>
-      <title>{title}</title>
+      <title>{props.titleAccess}</title>
```
Comment (sbegaudeau) [inline frontend/src/icons/SiriusIcon.tsx:26]: I think that you can remove the whole line, SvgIcon should do the job anyway

File: frontend/src/common/BrandingContext.tsx [hunk 2]
```diff
@@ -11,18 +11,18 @@
  *     Obeo - initial API and implementation
  *******************************************************************************/
 
+import { Help, SiriusIcon } from 'icons';
 import React, { useContext } from 'react';
-import { SiriusIcon, Help } from 'icons';
 import { Footer } from 'views/Footer';
 
-export const defaultIcon = <SiriusIcon title="" style={{ fill: 'var(--white)' }} />;
+export const defaultIcon = <SiriusIcon titleAccess="" style={{ fill: 'var(--white)' }} />;
```
Comment (sbegaudeau) [inline frontend/src/common/BrandingContext.tsx:18]: You can remove `titleAccess` entirely

File: frontend/src/icons/SiriusIcon.tsx [hunk 3]
```diff
@@ -10,23 +10,20 @@
...
 import React from 'react';
-import PropTypes from 'prop-types';
 
-const propTypes = {
-  title: PropTypes.string.isRequired,
-};
-
-export const SiriusIcon = ({ title, ...props }) => {
+export const SiriusIcon = (props: SvgIconProps) => {
+  debugger;
```
Comment (sbegaudeau) [inline frontend/src/icons/SiriusIcon.tsx:17]: Remove debugger;

File: frontend/src/navbar/NavigationBar.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'flex-end',
+  },
+}));
+
+export const NavigationBar = () => {
+  const classes = useNavigationbarStyles();
+  debugger;
```
Comment (sbegaudeau) [inline frontend/src/navbar/NavigationBar.tsx:47]: Remove debugger

File: frontend/src/navbar/NavigationBar.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Obeo.
```
Comment (sbegaudeau) [inline frontend/src/navbar/NavigationBar.tsx:2]: 2021 not 2019, 2020

File: frontend/src/navbar/UserStatusContextMenu.tsx [hunk 6]
```diff
@@ -31,7 +31,7 @@ export const UserStatusContextMenu = ({ x, y, userName, onLoggedOut, onClose })
       </div>
       <Separator />
       <a href="https://ocpmodeler.com/doc/welcome_help_center.html" rel="noopener noreferrer" target="_blank">
-        <Entry icon={<Help title="Help" />} label="Help" data-testid="help" />
+        <Entry icon={<Help titleAccess="Help" />} label="Help" data-testid="help" />
```
Comment (sbegaudeau) [inline frontend/src/navbar/UserStatusContextMenu.tsx:34]: Has the Help component been migrated to SvgIcon? If we went this way, we should probably just replace it with the regular MaterialUI help icon and probably remove the titleAccess here. It does not provide any meaningful information

File: frontend/src/stories/navbar/NavbarStory.tsx [hunk 7]
```diff
@@ -45,7 +44,7 @@ const LoggedInNavbarSection = () => {
     <div className={styles.section}>
       LoggedInNavbar
```
Comment (sbegaudeau) [inline frontend/src/stories/navbar/NavbarStory.tsx:45]: Not LoggedInNavbar anymore

File: frontend/src/common/BrandingContext.tsx [hunk 8]
```diff
@@ -11,18 +11,18 @@
...
 
 const defaultValue = {
   icon: defaultIcon,
   productName: 'Sirius Web',
   userStatus: (
     <a href="https://www.eclipse.org/sirius" rel="noopener noreferrer" target="_blank">
-      <Help style={{ fill: 'var(--white)', marginTop: 16, width: 32, height: 32 }} title="Help" />
+      <Help style={{ fill: 'var(--white)', marginTop: 16, width: 32, height: 32 }} titleAccess="Help" />
```
Comment (sbegaudeau) [inline frontend/src/common/BrandingContext.tsx:25]: See my other comment later on the Help icon

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: frontend/src/icons/Help.tsx [hunk 9]
```diff
@@ -19,16 +19,13 @@
...
 import React from 'react';
-import PropTypes from 'prop-types';
 
-const propTypes = {
-  title: PropTypes.string.isRequired,
-};
-
-export const Help = ({ title, ...props }) => {
+export const Help = (props: SvgIconProps) => {
+  debugger;
```
Comment (sbegaudeau) [inline frontend/src/icons/Help.tsx:26]: Please check that you don't commit `debugger;` instructions

File: frontend/src/icons/Help.tsx [hunk 10]
```diff
@@ -37,10 +34,8 @@ export const Help = ({ title, ...props }) => {
       aria-describedby="desc"
       role="img"
       {...props}>
-      <title>{title}</title>
+      <title>{props.titleAccess}</title>
```
Comment (sbegaudeau) [inline frontend/src/icons/Help.tsx:37]: Same comment as in the other icon, the whole line could be removed

File: frontend/src/common/BrandingContext.tsx (hunk 8 above)
Comment (gcoutable) [reply frontend/src/common/BrandingContext.tsx]: Done

File: frontend/src/navbar/UserStatusContextMenu.tsx (hunk 6 above)
Comment (gcoutable) [reply frontend/src/navbar/UserStatusContextMenu.tsx]: Done

File: frontend/src/stories/navbar/NavbarStory.tsx (hunk 7 above)
Comment (gcoutable) [reply frontend/src/stories/navbar/NavbarStory.tsx]: Done

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**